### DISCUSSION TITLE
state of being up to date after accepting a pull-request and branches

### DISCUSSION THREAD

**Discussion by ghost** [community/community#21677] [New to GitHub]
<p>Hello,</p>
<p>on a new github project I created 2 new branches next to master - “contrib” which is meant for contributions of other people and “remote” that is accepting my changes from at home. But when people fork master, do changes and send a pull-request it will end up in master, right? Or will there be a way to select in which branch the changes go? And lets say I merged everything into master will the other two branches be up to date automatically?</p>
<p>Thanks in advance,</p>
<p>P.</p>

**Comment by AndreaGriffiths11** [comment, marked as answer]
<p>Hi <a class="mention" href="/cleymspaze">@cleymspaze</a>,</p>
<p>Thank you for being here!  You can tell git when you want to checkout the branch with:</p>
<p><code>git checkout &lt;name&gt;</code></p>
<p>``</p>
<p>This will put you on the new branch before committing changes. With GitHub, you can deploy from a branch for final testing in production before merging to master. Also, publish your changes to the remote by using “git push” from time to time.</p>
<p>For instance your workflow may look something like this: </p>
<ol>
<li>Create a private branch (remote) off a public branch, in your case the contrib branch.</li>
<li>Regularly commit your work to this private branch.</li>
<li>Once your code is perfect, clean up its history.</li>
<li>Merge the cleaned-up branch back into the public branch.</li>
</ol>
<p>I hope this helps! I look foward to other folk advise on best practice here. For us the <a href="https://guides.github.com/introduction/flow/" rel="nofollow noopener">GitHub flow</a> is the norm.</p>
